ChangeLog.d/tests-common-code.txt

```diff
@@ -0,0 +1,11 @@
+Changes
+   * Test specific header files are now located in a dedicated 'include/test'
+     folder in 'tests' folder, not anymore directly in the 'tests' folder.
+   * An 'src' folder has been created in 'tests' folder to contain common test
+     code. The build systems have been modified to build the C files in that
+     'src' folder into an 'mbedtests' library that is linked to the test
+     suites.
+   * Test helper macros and functions from helpers.function, that are not
+     related to the unit test execution machinery have been moved to .c and
+     .h files located respectively in 'tests/src' and 'tests/include/test'
+     folders.
```

Comment:
I think the changelog entry is a bit too verbose, and all the changes should be in a single entry. The audience is people who build the tests, not people who maintain the test code. What we need to convey is that if you were using your own build instructions to build the tests, you need to update them. Suggestion:

> The unit tests now rely on header files in tests/include/test and source files in tests/src. When building with make or cmake, the files in tests/src are compiled to a libmbedtests library which is linked statically into each test executable (even when the product libraries are linked dynamically).

Reply:
I have adapted your proposal to the fact that there is no libmbedtests library anymore.